core/src/main/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraints.java

```diff
@@ -51,7 +51,8 @@ private static Set<Pathfinding.Range> getBlockedRanges(RollingStock stock, PathP
                 var blockingRanges = neutralSections.complement().subRangeSet(voltageInterval).asRanges();
 
                 for (var blockingRange : blockingRanges)
-                    res.add(new Pathfinding.Range(blockingRange.lowerEndpoint(), blockingRange.upperEndpoint()));
+                    if (blockingRange.lowerEndpoint() < blockingRange.upperEndpoint())
```

Comment:
0-length ranges can happen because of the way we build the `RangeSet`, I believe this was already the case before but it wasn't caught. 

But I've fixed it by using closed intervals instead when building the set, it makes more sense. I'll change that `if` into an `assert`.